src/AuthLDAP.php

```diff
@@ -3793,38 +3807,36 @@ public static function getDefault()
         return count($it) ? $it->current()['id'] : 0;
     }
 
+    /**
+     * @return void
+     */
     public function post_updateItem($history = true)
     {
-        /** @var \DBmysql $DB */
-        global $DB;
-
-        if (in_array('is_default', $this->updates, true) && (int) $this->input["is_default"] === 1) {
-            $DB->update(
-                static::getTable(),
-                ['is_default' => 0],
-                ['id' => ['<>', $this->input['id']]]
-            );
+        if (isset($this->fields["is_default"]) && $this->fields["is_default"] === 1) {
+            $this->removeDefaultFromOtherItems((int) $this->fields['id']);
         }
+
+        parent::post_updateItem($history);
     }
 
+    /**
+     * @return void
+     */
     public function post_addItem()
     {
-        /** @var \DBmysql $DB */
-        global $DB;
-
-        if (isset($this->fields['is_default']) && (int) $this->fields["is_default"] === 1) {
-            $DB->update(
-                static::getTable(),
-                ['is_default' => 0],
-                ['id' => ['<>', $this->fields['id']]]
-            );
+        if (isset($this->fields["is_default"]) && $this->fields["is_default"] === 1) {
+            $this->removeDefaultFromOtherItems((int) $this->fields['id']);
         }
+
+        // @todo notice the previous implementation did not call the parent, does it trigger side effect ? should it not be fixed ?
+        parent::post_addItem();
     }
 
     public function prepareInputForAdd($input)
     {
-        // If it's the first ldap directory then set it as the default directory
-        if (!self::getNumberOfServers()) {
+        // If there no other AuthLDAP or AuthMail, set it as default
+        // @todo the same behavior is not defined for AuthMail, it should be, maybe. Should I create an issue or is it not important ?
+        if (is_null(static::getDefaultAuth())) {
             $input['is_default'] = 1;
         }
```

Comment:
Honestly, I am not sure that forcing the first added server to be the default server is something we should keep. @orthagh, @flegastelois ?